patches/tModLoader/Terraria/ModLoader/ModNet.cs

```diff
@@ -227,12 +229,56 @@ internal static bool SyncClientMods(BinaryReader reader, out bool needsReload)
 
 			LocalMod matching = modFiles.FirstOrDefault(mod => header.Matches(mod.modFile));
 			if (matching != null) {
-				matching.Enabled = true;
+				if(clientMod != null) {
+					// Mod is enabled, but wrong version enabled
+					if(clientMod.Version > header.version) {
+						// TODO: Localize these messages once finalized
+						reloadRequiredExplanationEntries.Add(new ReloadRequiredExplanation(2, header.name, matching, $"[c/FFFACD:Switch] to v{header.version}\n(Temporarily downgrade from v{clientMod.Version})"));
+					}
+					else {
+						reloadRequiredExplanationEntries.Add(new ReloadRequiredExplanation(2, header.name, matching, $"[c/FFFACD:Switch] to v{header.version}\n(Temporarily upgrade from v{clientMod.Version})"));
+					}
+				}
+				else {
+					toEnable.Add(matching.Name);
+					reloadRequiredExplanationEntries.Add(new ReloadRequiredExplanation(3, header.name, matching, $"[c/98FB98:Enable]")); 
+				}
```

Comment:
As recommended below, perhaps we can split the functional code (`toEnable.Add(matching.Name)`) from the UI display message creation